modules/atomic-widgets/schema/constraints/enum.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace Elementor\Modules\AtomicWidgets\Schema\Constraints;
+
+use Elementor\Core\Utils\Collection;
+use Elementor\Modules\AtomicWidgets\Schema\Prop_Constraint;
+use Elementor\Utils;
+
+class Enum extends Prop_Constraint {
+	private array $allowed_values;
+
+	public function get_type(): string {
+		return 'enum';
+	}
+
+	public function get_value(): array {
+		return $this->allowed_values;
+	}
+
+	public function __construct( array $allowed_values ) {
+		if ( ! $this->all_are_strings( $allowed_values ) ) {
+			Utils::safe_throw( 'All values in an enum must be strings.' );
+		}
+
+		$this->allowed_values = $allowed_values;
+	}
+
+	public static function make( array $allowed_values ): self {
+		return new static( $allowed_values );
+	}
+
+	public function validate( $value ): void {
```

Comment:
Do we need to return something from this?

Reply:
no